core/services/attributes_generator.rb

```diff
@@ -0,0 +1,188 @@
+# frozen_string_literal: true
+
+require_relative 'configuration_generator'
+require_relative 'product_attributes'
+require_relative '../out'
+
+# The class provides methods for generating attributes for configuration generator
+module AttributesGenerator
+  # Make product config and recipe name for install it to the VM.
+  #
+  # @param product [Hash] parameters of product to configure from configuration file
+  # @param box [String] name of the box
+  # @return [Result::Base] recipe name and product config in format { recipe: String, config: Hash }.
+  # rubocop:disable Metrics/MethodLength
+  def self.make_product_config_and_recipe_name(product, box, log, env)
```

Comment:
Please do not pass env as a parameter to any method. If a service depends on something, then provide it directly.

You should consider to make AttributeGenerator a class if it depends on so many external services.

Reply:
This will shorten the number of arguments passed to the method.